dataprofiler/profilers/profile_builder.py

```diff
@@ -1431,53 +1436,61 @@ def _update_row_statistics(self, data, sample_ids=None):
             self.row_has_null_count = len(null_in_row_count)
             self.row_is_null_count = len(null_rows)
 
-    def _get_correlation(self, clean_samples):
+    def _get_correlation_and_properties(self, data):
         """
-        Calculate correlation matrix on the cleaned data.
+        Calculate correlation matrix and necessary batch properties on the data.
 
-        :param clean_samples: the input cleaned dataset
-        :type clean_samples: dict()
+        :param data: the input dataset
+        :type data: pd.Dataframe
         """
         columns = self.options.correlation.columns
         clean_column_ids = []
         if columns is None:
             for idx in range(len(self._profile)):
-                if (self._profile[idx].profile['data_type'] not in ['int', 'float']
-                        or self._profile[idx].null_count > 0):
-                    clean_samples.pop(idx)
-                else:
+                if self._profile[idx].profile['data_type'] in ['int', 'float']:
                     clean_column_ids.append(idx)
 
-        data = pd.DataFrame(clean_samples)
-        data = data.apply(pd.to_numeric, errors='coerce')
+        # Exclude all rows with NaN in numeric columns
+        # and add dropped-NaN stats to properties for
+        # future updating
+        data_dropped = data.dropna(subset=data.columns[clean_column_ids])
```

Comment:
NA values dropped should have already been determined via `clean_data_and_get_base_stats` which is called in the update chunk method prior to this function. Can we pass that info as well so that we don't recalculate?

Reply:
@JGSweets I thought that does it on a per-column basis. In this case, if one column has a NaN, the rest of the columns should have their corresponding values dropped as well even if there's non-null values. For example, if `col1=[NaN, 3, 5]` and `col2=[1,2,3]` , we still want to drop the `1` from `col2` since there's a `NaN` in `col1`.

Reply:
Agree with Andrew. with this option of nan processing for correlation, I don't think we can use the clean samples obtained from the previous step, as the columns' indices are non-overlap